File: backend/app/schemas/irrigation.py
from pydantic import BaseModel
from typing import List, Optional

class FarmSummary(BaseModel):
    crop: str
    soil_type: str
    irrigation_type: str
    farm_size: str
    location: str

class WeatherSummary(BaseModel):
    temperature: float
    humidity: int
    rainfall: float
    wind_speed: float
    rain_probability: int
    expected_rainfall_48h: float
    weather_condition: str

class IrrigationHistoryItem(BaseModel):
    id: int
    status: str
    status_label: str
    recommendation: str
    priority: str
    created_at: str

class IrrigationAdvisoryResponse(BaseModel):
    success: bool
    status: str
    status_label: str
    priority: str
    recommendation: str
    reasons: List[str]
    farm_summary: FarmSummary
    weather_summary: Optional[WeatherSummary] = None
    missing_parameters: List[str] = []
    history: List[IrrigationHistoryItem] = []
    generated_at: Optional[str] = None

File: frontend/src/types/irrigation.ts
export interface FarmSummary {
  crop: string;
  soil_type: string;
  irrigation_type: string;
  farm_size: string;
  location: string;
}

export interface WeatherSummary {
  temperature: number;
  humidity: number;
  rainfall: number;
  wind_speed: number;
  rain_probability: number;
  expected_rainfall_48h: number;
  weather_condition: string;
}

export interface IrrigationHistoryItem {
  id: number;
  status: string;
  status_label: string;
  recommendation: string;
  priority: string;
  created_at: string;
}

export interface IrrigationAdvisoryResponse {
  success: boolean;
  status: 'delay' | 'needed' | 'check_moisture' | 'insufficient_data';
  status_label: string;
  priority: 'Low' | 'Medium' | 'High';
  recommendation: string;
  reasons: string[];
  farm_summary: FarmSummary;
  weather_summary?: WeatherSummary | null;
  missing_parameters: string[];
  history: IrrigationHistoryItem[];
  generated_at?: string | null;
}
